src/mrinufft/operators/off_resonance.py

```diff
@@ -0,0 +1,325 @@
+"""Off Resonance correction Operator wrapper.
+
+Based on the implementation of Guillaume Daval-Frérot in pysap-mri:
+https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/operators/fourier/orc_wrapper.py
+"""
+
+import math
+import numpy as np
+
+from .._utils import get_array_module
+
+from .base import FourierOperatorBase
+from .interfaces.utils import is_cuda_array
+
+CUPY_AVAILABLE = True
+try:
+    import cupy as cp
+except ImportError:
+    CUPY_AVAILABLE = False
+
+
+TORCH_AVAILABLE = True
+try:
+    import torch
+except ImportError:
+    TORCH_AVAILABLE = False
```

Comment:
I think you can pull `CUPY_AVAILABLE` and `TORCH_AVAILABLE` from `.base`:
https://github.com/mind-inria/mri-nufft/blob/aad8c0e77f6c059d139e2ba37ce7f78e0914a787/src/mrinufft/operators/base.py#L22-L34

 (Note that TORCH_AVAILABLE is called `AUTOGRAD_AVAILBLE`